compiler/src/iree/compiler/Codegen/Common/TensorDynamicDimAnalysis.cpp

```diff
@@ -0,0 +1,236 @@
+// Copyright 2024 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "iree/compiler/Codegen/Common/TensorDynamicDimAnalysis.h"
+#include "iree/compiler/Dialect/Flow/IR/FlowOps.h"
+#include "iree/compiler/Dialect/Util/Analysis/IntegerDivisibilityAnalysis.h"
+#include "llvm/Support/Debug.h"
+#include "mlir/Analysis/DataFlow/DeadCodeAnalysis.h"
+#include "mlir/Analysis/DataFlow/IntegerRangeAnalysis.h"
+#include "mlir/Dialect/Tensor/IR/Tensor.h"
+#include "mlir/Interfaces/DestinationStyleOpInterface.h"
+
+#define DEBUG_TYPE "iree-codegen-dynamic-dim-analysis"
+
+namespace mlir::iree_compiler {
+
+//===---------------------------------------------------------------------===//
+// Helper function to update tensor dynamic dimension info
+//===---------------------------------------------------------------------===//
+
+static void
+updateRangeInfo(TensorDynamicDimAnalysis::TensorDimRangeInfo &rangeInfo,
```

Comment:
`updateRangeInfo` -> `initializeRangeInfo` because overwriting existing info is not allowed.

Reply:
`rangeInfo` is being updated here for the current value?

Reply:
I am reading this as initializing the range info for the given value. Initialize is clearer to me but I won't block then.